src/audio/wav64.c

```diff
@@ -165,7 +181,15 @@ void raw_waveform_read(samplebuffer_t *sbuf, int base_rom_addr, int wpos, int wl
 static void waveform_read(void *ctx, samplebuffer_t *sbuf, int wpos, int wlen, bool seeking) {
 	wav64_t *wav = (wav64_t*)ctx;
 	int bps = (wav->wave.bits == 8 ? 0 : 1) + (wav->wave.channels == 2 ? 1 : 0);
-	raw_waveform_read(sbuf, wav->rom_addr, wpos, wlen, bps);
+	int start_position;
+	if (seeking) {
+		start_position = lseek(wav->current_fd, 0, SEEK_CUR);
+		lseek(wav->current_fd, wav->base_offset + (wpos << bps), SEEK_SET);
+	}
+	raw_waveform_read(sbuf, wav->current_fd, wpos, wlen, bps);
+	if (seeking) {
+		lseek(wav->current_fd, start_position, SEEK_SET);
+	}
```

Comment:
There is no need to seek back to the previous position when `seeking` is `true`. This also means you don't need the initial `lseek` to read the current position.